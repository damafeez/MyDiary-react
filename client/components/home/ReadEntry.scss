.read-entry {
  flex: 1 0 25rem;
  h2{
    font-weight: 300;
    font-size: 3.5rem;
    margin-bottom: 1.5rem;
  }
  p.date{
    font-weight: 300;
    font-size: 1.4rem;
    color: #43a047;
    margin-bottom: 2rem;
    border-bottom: 3px solid;
    display: inline-block;
    text-transform: uppercase;
  }
  p.body {
    font-size: 1.5rem;
    letter-spacing: .3px;
    line-height: 1.4;
    color: var(--lighter-text-color);
  }
  @media screen and (min-width: 550px) {
    flex: 1 0 30rem;
  }
  @media screen and (min-width: 800px) {
    flex: 1 0 40rem;
  }
  @media screen and (min-width: 1100px) {
    flex: 1 0 45rem;
  }
}