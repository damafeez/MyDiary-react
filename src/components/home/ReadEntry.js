import React, { Component } from 'react';
import './ReadEntry.scss';

export default class AddEntry extends Component {
  state = {
    title: '',
    body: '',
  }

  render() {
    const { title, body } = this.state;
    return (
      <section className="read-entry">
        <h2>This is a simple title</h2>
        <p className="date">JAN 7</p>
        <p className="body">Lorem ipsum dolor sit amet consectetur, adipisicing elit. Inventore voluptate, ea ex iste, provident beatae, aperiam optio corrupti accusantium veritatis natus suscipit est? Expedita ab veritatis esse cum asperiores velit.</p>
      </section>
    );
  }
}
