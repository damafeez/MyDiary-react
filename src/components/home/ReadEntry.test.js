import React from 'react';
import ReadEntry from './ReadEntry';

const setup = (overrideProps) => {
  const props = {
    currentIndex: 0,
    entries: [{
      id: 1, title: 'this title', body: 'this body', created: (new Date(5000)),
    }],
  };
  const wrapper = shallow(<ReadEntry {...props} {...overrideProps} />);
  return { props, wrapper };
};

it('matches snapshot', () => {
  const { wrapper } = setup();
  expect(wrapper).toMatchSnapshot();
});
it('matches snapshot:::loading', () => {
  const { wrapper } = setup({ entries: [], loading: true });
  expect(wrapper).toMatchSnapshot();
});
it('matches snapshot:::error', () => {
  const { wrapper } = setup({ entries: [], error: true });
  expect(wrapper).toMatchSnapshot();
});
it('matches snapshot:::entries:[]', () => {
  const { wrapper } = setup({ entries: [] });
  expect(wrapper).toMatchSnapshot();
});
