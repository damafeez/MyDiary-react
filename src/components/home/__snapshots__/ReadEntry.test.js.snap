// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentIndex={0}
    entries={
      Array [
        Object {
          "body": "this body",
          "created": 1970-01-01T00:00:05.000Z,
          "id": 1,
          "title": "this title",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          this title
        </h2>,
        <p
          className="date"
        >
          JAN 1
        </p>,
        <p
          className="body"
        >
          this body
        </p>,
      ],
      "className": "read-entry",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this title",
        },
        "ref": null,
        "rendered": "this title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "JAN 1",
          "className": "date",
        },
        "ref": null,
        "rendered": "JAN 1",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this body",
          "className": "body",
        },
        "ref": null,
        "rendered": "this body",
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            this title
          </h2>,
          <p
            className="date"
          >
            JAN 1
          </p>,
          <p
            className="body"
          >
            this body
          </p>,
        ],
        "className": "read-entry",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this title",
          },
          "ref": null,
          "rendered": "this title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "JAN 1",
            "className": "date",
          },
          "ref": null,
          "rendered": "JAN 1",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this body",
            "className": "body",
          },
          "ref": null,
          "rendered": "this body",
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot:::entries:[] 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentIndex={0}
    entries={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          You have not added any entry to your diary
        </h2>,
        <p
          className="date"
        />,
        <p
          className="body"
        >
          Please click the green  button at bottom left to get started
        </p>,
      ],
      "className": "read-entry",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have not added any entry to your diary",
        },
        "ref": null,
        "rendered": "You have not added any entry to your diary",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "date",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please click the green  button at bottom left to get started",
          "className": "body",
        },
        "ref": null,
        "rendered": "Please click the green  button at bottom left to get started",
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            You have not added any entry to your diary
          </h2>,
          <p
            className="date"
          />,
          <p
            className="body"
          >
            Please click the green  button at bottom left to get started
          </p>,
        ],
        "className": "read-entry",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have not added any entry to your diary",
          },
          "ref": null,
          "rendered": "You have not added any entry to your diary",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "date",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please click the green  button at bottom left to get started",
            "className": "body",
          },
          "ref": null,
          "rendered": "Please click the green  button at bottom left to get started",
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot:::error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentIndex={0}
    entries={Array []}
    error={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          true
        </h2>,
        <p
          className="date"
        />,
        <p
          className="body"
        />,
      ],
      "className": "read-entry",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": true,
        },
        "ref": null,
        "rendered": true,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "date",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "body",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            true
          </h2>,
          <p
            className="date"
          />,
          <p
            className="body"
          />,
        ],
        "className": "read-entry",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": true,
          },
          "ref": null,
          "rendered": true,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "date",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "body",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot:::loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentIndex={0}
    entries={Array []}
    loading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Loading....
        </h2>,
        <p
          className="date"
        />,
        <p
          className="body"
        />,
      ],
      "className": "read-entry",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading....",
        },
        "ref": null,
        "rendered": "Loading....",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "date",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "body",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Loading....
          </h2>,
          <p
            className="date"
          />,
          <p
            className="body"
          />,
        ],
        "className": "read-entry",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading....",
          },
          "ref": null,
          "rendered": "Loading....",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "date",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "body",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
